src/l500/l500-private.h

```diff
@@ -333,14 +333,26 @@ namespace librealsense
         {
             orientation orient;
             resolutions_depth resolution;
+
+            size_t get_expected_size(const std::vector<uint8_t> & raw_data_vec) const 
+            {
+                size_t expected_size(0);
+                const auto raw = reinterpret_cast<const intrinsic_depth *>(raw_data_vec.data());
+
+                expected_size += sizeof(raw->orient);
+                // Get full maximum size of the resolution array and deduct the unused resolutions size from it.
+                expected_size += sizeof(raw->resolution);
```

Comment:
Instead of adding up the individual fields, why not do sizeof( *this ) - ...?

Reply:
I thought it is more readable and it's only 2 fields, if you prefer "*this" I can change it.
your call..

Reply:
I am concerned that another field will get added and then this calculation will be wrong.
If we know the variable size depends on only one variable, let's make the code dependent on only one variable.

Reply:
OK, Will be updated.